src/API/Plugin.php

```diff
@@ -11,7 +11,16 @@ class Plugin extends Client
     const SETTING_DEFAULT_SETTINGS = 'default_settings';
     const SETTING_IP_REWRITE = 'ip_rewrite';
     const SETTING_PROTOCOL_REWRITE = 'protocol_rewrite';
-    const PLUGIN_SPECIFIC_CACHE = 'plugin_specific_cache';
+    const SETTING_PLUGIN_SPECIFIC_CACHE = 'plugin_specific_cache';
+
+    public static function getPluginSettingsKeys() {
+        return array(
+            self::SETTING_DEFAULT_SETTINGS,
+            self::SETTING_IP_REWRITE,
+            self::SETTING_PROTOCOL_REWRITE,
+            self::SETTING_PLUGIN_SPECIFIC_CACHE
+        );
+    }
```

Comment:
**Problem:**
When a new plugin action needs to be added we need to add new keys to `getPluginSettingsKeys` but that should be avoided until we add the same functionality to most of our plugins. 

**Solution**
Making `getPluginSettingsKeys` abstract and by default supporting these shared keys is a solution. Also we can make a public function allowing us to add a key (though I'm not sure if there's an appropriate place to use that function in the plugin code)